clients/src/main/java/org/apache/kafka/clients/consumer/internals/StreamsRebalanceData.java

```diff
@@ -355,6 +357,9 @@ public Map<String, Subtopology> subtopologies() {
         return subtopologies;
     }
 
+    public int endpointInformationEpoch() {
+        return endpointInformationEpoch;
```

Comment:
`StreamsRebalanceData` is used as an interface between the application thread and the streams thread. The data here should be immutable or otherwise threadsafe. But I'm not sure we need even access this epoch from the application thread? I'd consider moving the client-side epoch to `StreamsGroupHeartbeatManager.HeartbeatState`

Reply:
I'm going to disagree here, `StreamsRebalanceData` already contains related metadata `HostInfo`  and  `StreamsRebalanceData.partitionsByHost` cf [d6b6952d4](https://github.com/apache/kafka/commit/d6b6952d486894ddf1099c1ad42911bbaccb1533) So I think the `endpointEpoch` is ok here.  I've updated it to an `AtomicInteger` for thread safety.  Having said that, if feel otherwise, I'll move it to the `StreamsGroupHearbeatRequestManager.HeartbeatState`

Reply:
> the application thread and the streams thread

What is the "application thread" -- do you mean the thread that is using IQ API -- for this case, I am wondering why the application would need to access `StreamsRebalanceData` -- it seems to be some internal thing? As a matter of fact, this class is in a package called `...internal...`.

"streams thread" should literally be `StreamThread` right?

Reply:
Sorry, typo. I meant to say "interface between applciation thread (= stream thread) and consumer networking thread". `StreamsRebalanceData` contains `partitionsByHost` because the streams thread needs to access it. But the `endpointEpoch` is not needed by the streams thread, afaik.

In the end it's an internal class right now, so maybe it does not matter too much. However, if we plan to provide a public interface for wrapping / mocking the streams consumer, this will mean that we have to make `StreamsRebalanceData` a public interface. So I would like to keep it as simple as poissible.

Reply:
OK fair enough I'll move `endpointEpoch` to `StreamsGroupHeartbeatRequestManager.HeartbeatRequestState`